### DISCUSSION TITLE
Context managers in requests?

### DISCUSSION THREAD

**Discussion by controlado** [aio-libs/aiohttp#7321] [Q&A]
I know that the use of context managers in requests is not mandatory, but I wonder why I should use them?

- Is there a performance gain?
- Is it just a matter of code readability?

```py
async def get_something(session):
    response = await session.get(...)
```

```py
async def get_something(session):
    async with session.get(...) as response:
        ...
```

**Comment by Dreamsorcerer** [comment]
To ensure that the connection is closed properly.
